tests/tensorflow/test_tensorflow_estimator_export.py

```diff
@@ -0,0 +1,251 @@
+import pandas as pd
+import pandas.testing
+import tensorflow as tf
+import iris_data_utils
+
+import mlflow
+import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
+from mlflow import pyfunc
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.tensorflow import _TF2Wrapper
+from mlflow.utils.conda import get_or_create_conda_env
+from mlflow.pyfunc.backend import _execute_in_conda_env
+
+from tests.helper_functions import (
+    pyfunc_serve_and_score_model,
+    assert_pandas_dataframe_almost_equal,
+    assert_array_almost_equal,
+)
+
+
+@pytest.fixture
+def tf_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(conda_env, additional_pip_deps=["tensorflow", "pytest"])
+    return conda_env
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(str(tmpdir), "model")
+
+
+ModelDataInfo = collections.namedtuple(
+    "ModelDataInfo",
+    [
+        "inference_df",
+        "expected_results_df",
+        "raw_results",
+        "raw_df",
+        "run_id",
+    ],
+)
+
+
+def save_or_log_tf_model_by_mlflow128(tmpdir, model_type, task_type, model_path=None):
+    tf_tests_dir = os.path.dirname(__file__)
+    conda_env = get_or_create_conda_env(os.path.join(tf_tests_dir, "mlflow-128-tf-23-env.yaml"))
+    output_data_file_path = os.path.join(tmpdir, "output_data.pkl")
+    tracking_uri = mlflow.get_tracking_uri()
+    exec_py_path = os.path.join(tf_tests_dir, "save_tf_estimator_model.py")
+    mlflow_repo_path = os.path.dirname(os.path.dirname(tf_tests_dir))
+
+    _execute_in_conda_env(
+        conda_env,
+        f"python {exec_py_path} {tracking_uri} {mlflow_repo_path} {model_type} {task_type} "
+        f"{'--save_path ' + model_path if model_path else ''}",
+        install_mlflow=False,
+    )
+    with open(output_data_file_path, "rb") as f:
+        return ModelDataInfo(*pickle.load(f))
+
+
+def save_or_log_keras_model_by_mlflow128(tmpdir, task_type, save_as_type, save_path=None):
+    tf_tests_dir = os.path.dirname(__file__)
+    conda_env = get_or_create_conda_env(os.path.join(tf_tests_dir, "mlflow-128-tf-23-env.yaml"))
+    output_data_file_path = os.path.join(tmpdir, "output_data.pkl")
+    tracking_uri = mlflow.get_tracking_uri()
+    exec_py_path = os.path.join(tf_tests_dir, "save_keras_model.py")
+
+    _execute_in_conda_env(
+        conda_env,
+        f"python {exec_py_path} {tracking_uri} {task_type} {save_as_type} "
+        f"{'--save_path ' + save_path if save_path else ''}",
+        install_mlflow=False,
+    )
+    with open(output_data_file_path, "rb") as f:
+        inference_df, expected_results_df, run_id = pickle.load(f)
+        return ModelDataInfo(
+            inference_df=inference_df,
+            expected_results_df=expected_results_df,
+            raw_results=None,
+            raw_df=None,
+            run_id=run_id,
+        )
+
```

Comment:
This should go in `test_keras_model_export` along with https://github.com/mlflow/mlflow/pull/6530/files#r974963152